src/v/utils/fragmented_vector.h

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2020 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+
+#include "vassert.h"
+
+#include <cstddef>
+#include <vector>
+
+/**
+ * A very very simple fragmented vector that provides random access like a
+ * vector, but does not store its data in contiguous memory.
+ *
+ * There is no reserve method because we allocate a full fragment at a time.
```

Comment:
Maybe you can explain why "allocating a full fragment at a time" is at the core of this solution. In the case we implemented a reserve method, creating all fragments at once, would that mean the allocator is more likely to place all of these segments in a contiguous area of memory?

Reply:
fixed!